acceptance/openstack/blockstorage/v3/quotaset_test.go

```diff
@@ -40,6 +41,58 @@ func TestQuotasetGetUsage(t *testing.T) {
 	tools.PrintResource(t, quotaSetUsage)
 }
 
+var UpdateQuotaOpts = quotasets.UpdateOpts{
+	Volumes:            gophercloud.IntToPointer(100),
+	Snapshots:          gophercloud.IntToPointer(200),
+	Gigabytes:          gophercloud.IntToPointer(300),
+	PerVolumeGigabytes: gophercloud.IntToPointer(50),
+	Backups:            gophercloud.IntToPointer(2),
+	BackupGigabytes:    gophercloud.IntToPointer(300),
+}
+
+var UpdatedQuotas = quotasets.QuotaSet{
+	Volumes:            100,
+	Snapshots:          200,
+	Gigabytes:          300,
+	PerVolumeGigabytes: 50,
+	Backups:            2,
+	BackupGigabytes:    300,
+}
+
+func TestQuotasetUpdate(t *testing.T) {
+	client, projectID := getClientAndProject(t)
+
+	// save original quotas
+	orig, err := quotasets.Get(client, projectID).Extract()
+	th.AssertNoErr(t, err)
+
+	defer func() {
+		restore := quotasets.UpdateOpts{}
+		FillUpdateOptsFromQuotaSet(*orig, &restore)
+
+		_, err = quotasets.Update(client, projectID, restore).Extract()
+		th.AssertNoErr(t, err)
+	}()
+
+	// test Update
+	resultQuotas, err := quotasets.Update(client, projectID, UpdateQuotaOpts).Extract()
+	th.AssertNoErr(t, err)
+	th.AssertEquals(t, UpdatedQuotas, *resultQuotas)
+
+	// We dont know the default quotas, so just check if the quotas are not the
+	// same as before
+	newQuotas, err := quotasets.Get(client, projectID).Extract()
+	th.AssertNoErr(t, err)
+	if newQuotas.Volumes != resultQuotas.Volumes {
+		t.Fatalf(
+			fmt.Sprintf("Failed to update quotas\n\texpected: %d\t actual: %d",
+				resultQuotas.Volumes,
+				newQuotas.Volumes,
+			),
+		)
```

Comment:
It might be easier to structure this like:

```go
err := fmt.Errorf("Failed to ...")
t.Fatal(err)
```

But in this case, you can simplify it down to:

```go
newQuotas, err := quotasets.Get(client, projectID).Extract()
th.AssertNoErr(t, err)
th.AssertEquals(t, resultQuotas.Volumes, newQuota.Volumes)
```

`th.AssertEquals` will error and print out an almost identical statement when the two items differ.